### ISSUE TITLE
[Uninstall] Clicking Remove on Windows

### ISSUE THREAD

**Issue by regcs** [regcs/AliceLG-beta#13]
When removing the add-on, a traceback error occurs on Windows machines:

![bug-uninstall](https://user-images.githubusercontent.com/3891338/122129085-e249ad80-ce35-11eb-92d1-f107247df321.png)

It's unclear at the moment, if that is still true for newer versions of the add-on.

**Comment by omgitsraven** [comment]
Just happened to me too. Is there a recommended way to uninstall? (Or more specifically, to install a new version? Is it safe to just install on top, or does the old one need to be removed somehow first?)

**Comment by omgitsraven** [comment]
(For the record: I relaunched Blender, and the 'remove' button worked right on my second attempt.)

**Comment by regcs** [comment]
@omgitsraven Sorry for not responding to this so long. It should be no mature issue to remove it like that. 

Could you do me a favor and check, if this issue still persists with Alice/LG 2.0.4?

**Comment by omgitsraven** [comment]
Uninstalling Alice 2.0.4 from Blender 2.93.7 gives this error: 
![image](https://user-images.githubusercontent.com/8540042/146683909-dc51fc6a-5c2b-4a78-8749-cf50463033a4.png)

After receiving that error, relaunching Blender lets it uninstall normally.  (I had also relaunched once already after installing, btw.)

**Comment by regcs** [comment]
@omgitsraven Thanks for responding so quickly. Could you also provide the `alice-lg.log` and `pylightio.log` files from the /AliceLG/logs/ folder?

**Comment by omgitsraven** [comment]
What's the rest of the path to that folder?

**Comment by regcs** [comment]
It depends on the system. Basically in the folder where all the Blender add-ons are installed:

Windows: `%USERPROFILE%\AppData\Roaming\Blender Foundation\Blender\2.93\Scripts\addons`
macOS: `/Users/$USER/Library/Application Support/Blender/2.93/Scripts/addons`

**Comment by omgitsraven** [comment]
Well here's something interesting.  I was about to say "there's no `AliceLG` folder there, just `looking_glass_tools` which only contains `.py` files", because I saw this:
![image](https://user-images.githubusercontent.com/8540042/146685914-ff982201-2393-42da-85e8-d380e18c17ae.png)
and I'm just used to totally ignoring `__MACOSX` folders.
But in the interest of thoroughness, I opened it up, and there it is:
![image](https://user-images.githubusercontent.com/8540042/146685961-bf3e0ca0-6aaf-4acc-a2ea-155726fdaf1c.png)
the AliceLG/logs folder only contains this, however:
![image](https://user-images.githubusercontent.com/8540042/146686010-89074027-e168-419f-b492-ea5d5df88455.png)
I guess uninstalling deleted all these files?  Should I send a copy of what the files are like while it's still installed? (Or maybe, between the first and second attempt at uninstallation?)

**Comment by regcs** [comment]
Ah, yeah ... sure ... after clicking the remove button the files will be gone. The `_MACOSX` folder can be ignored. After installing Alice/LG the actual `AliceLG` folder should not be in `_MACOSX` but in the addons folder.

One question then: When you click "Remove" button, the add-on disappears but the error is shown. Or is it rather: The error is shown and when you close the error message, the addon is removed?

**Comment by omgitsraven** [comment]
Neither:

1. I click 'remove'.
2. The error appears, but the plugin IS STILL INSTALLED.
3. I close Blender, then relaunch it.
4. The plugin is STILL installed.
5. I click 'remove'.
6. Now the plugin uninstalls.

**Comment by regcs** [comment]
Oh, ok ... and in step 4, is the plugin disabled or still enabled?

Anyway, then the best thing would be to get the log-files directly after step 2.

**Comment by omgitsraven** [comment]
The plugin becomes deactivated as soon as I click 'remove' in step 1.
Here are the logs after step 2:
[alice-lg.log](https://github.com/regcs/AliceLG/files/7746636/alice-lg.log)
[pylightio.log](https://github.com/regcs/AliceLG/files/7746638/pylightio.log)

**Comment by regcs** [comment]
Thanks for the log files! 

I will do some research on that to see if that can be fixed on the add-on level.

**Comment by regcs** [comment]
After a little research it seems that this is a general bug, which can not really be solved on the addon level and has to do with third-party compiled libraries used by the addon:

https://developer.blender.org/T77837

**Workaround** 
The only workaround at the moment would be to give the user the following uninstall instructions:

1. Disabled the addon.
2. Restart Blender.
3. Remove the addon.

**Comment by omgitsraven** [comment]
Good enough for me, if that's all that can be done. (Is there any way to hook into the 'remove' button and make those instructions pop up if it's clicked while the addon is still active?)

**Comment by regcs** [comment]
I will try to show some instructions, but I am not sure if Blender allows this via a pop up. But I will have a look.

There is one thing I would like to try though. I found some ways to manually unload libraries from python. Maybe you can test, if this works by installing the following version of Alice/LG:

https://github.com/regcs/AliceLG/archive/refs/heads/alice-lg-2.0.5.zip

**Comment by omgitsraven** [comment]
Couldn't install it at all; when I clicked the checkbox to activate it, it gave me this:
![image](https://user-images.githubusercontent.com/8540042/146959216-5720158b-753c-49f8-a28f-c5a80794ec2a.png)

**Comment by regcs** [comment]
Yeah, sorry. Please download the file again. I renamed the folder in the zip.

**Comment by omgitsraven** [comment]
Same problem as before:

![image](https://user-images.githubusercontent.com/8540042/146960200-ea7d63c8-6b45-474f-b4ef-eafda70e1bb5.png)

**Comment by regcs** [comment]
Please always provide the log files, because this error is not showing the actual cause. Thanks!

**Comment by omgitsraven** [comment]
[alice-lg.log](https://github.com/regcs/AliceLG/files/7756637/alice-lg.log)
[pylightio.log](https://github.com/regcs/AliceLG/files/7756638/pylightio.log)

**Comment by regcs** [comment]
Indeed the same error. Ok, I guess then there is nothing I can do about it for now. Thanks for your help in narrowing this down, @omgitsraven.

**Comment by regcs** [comment]
Note for myself:

Try if implementing the following in Blender's removal routine (`scripts\startup\bl_operators\userpref.py`) would help.

```
import os, stat
import shutil

def remove_readonly(func, path, _):
   "Clear the readonly bit and reattempt the removal"
os.chmod(path, stat.S_IWRITE)
func(path)

shutil.rmtree(directory, onerror = remove_readonly)
```

**Comment by regcs** [comment]
Submitted a patch for Blender which should take care of the issue: 

https://developer.blender.org/D13685
